From 15c4495e47afe5b4db8955dbacfeadce1b9410f2 Mon Sep 17 00:00:00 2001
From: Jonathan Helmus <jjhelmus@gmail.com>
Date: Tue, 30 Oct 2018 14:26:59 -0500
Subject: [PATCH 4/8] openssl 1.1.1 support on windows

---
 plugins/NTMakefile | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/plugins/NTMakefile b/plugins/NTMakefile
index ea2e0889..ae66503a 100755
--- a/plugins/NTMakefile
+++ b/plugins/NTMakefile
@@ -62,6 +62,8 @@ compat_objs = getaddrinfo.obj getnameinfo.obj
 
 common_objs = plugin_common.obj $(compat_objs)
 
+crypto_compat_objs = crypto-compat.obj
+
 saslANONYMOUS_objs = anonymous.obj anonymous_init.obj $(common_objs)
 saslANONYMOUS_out = saslANONYMOUS.dll saslANONYMOUS.exp saslANONYMOUS.lib
 
@@ -86,7 +88,7 @@ saslNTLM_out = saslNTLM.dll saslNTLM.exp saslNTLM.lib
 saslGSSAPI_objs = gssapi.obj gssapiv2_init.obj $(common_objs)
 saslGSSAPI_out = saslGSSAPI.dll saslGSSAPI.exp saslGSSAPI.lib
 
-saslSRP_objs = srp.obj srp_init.obj $(common_objs)
+saslSRP_objs = srp.obj srp_init.obj $(common_objs) $(crypto_compat_objs)
 saslSRP_out = saslSRP.dll saslSRP.exp saslSRP.lib
 
 saslOTP_objs = otp.obj otp_init.obj $(common_objs)
@@ -192,6 +194,10 @@ install: $(PLUGINS)
 
 all-recursive : $(PLUGINS)
 
+crypto-compat.c: ..\common\crypto-compat.c
+	xcopy /D /Y ..\common\crypto-compat.c .
+	xcopy /D /Y ..\common\crypto-compat.h .
+
 #Add /pdb: option?
 
 saslANONYMOUS.dll: $(saslANONYMOUS_objs) saslANONYMOUS.res
-- 
2.23.0

